src/cookbook/gradual-compositionapi-migration.md

```diff
@@ -0,0 +1,258 @@
+
+```js
+export function useToggle() {
+  const state = reactive({
+    isShowing: false
+  })
+
+  function toggleShow() {
+    state.isShowing = !state.isShowing
+  }
+
+  return {
+    ...toRefs(state),
+    toggleShow
+  }
+}
+```
+
+And we can refactor our component as follows:
+
+```html
+<template>
+  <div>
+    <h3>Let's trigger this here modal!</h3>
+
+    <button @click="toggleShow">
+      <span v-if="isShowing">Hide child</span>
+      <span v-else>Show child</span>
+    </button>
+
+    <div v-if="isShowing" class="modal">
+      <h2>Here I am!</h2>
+      <button @click="toggleShow">Close</button>
+    </div>
+  </div>
+</template>
+```
+
+```js
+<script>
+  import { useToggle } from "@/composables/useToggle.js";
+
+  export default {
+    setup() {
+      const { isShowing, toggleShow } = useToggle();
+
+      return {
+        isShowing,
+        toggleShow
+      };
+    },
+  }
+</script>
+```
+
+<iframe src="https://codesandbox.io/embed/refactor-mixin-to-composition-api-hnloh?fontsize=14&hidenavigation=1&theme=dark"
+     style="width:100%; height:500px; border:0; border-radius: 4px; overflow:hidden;"
+     title="Refactor Mixin to Composition API"
+     allow="accelerometer; ambient-light-sensor; camera; encrypted-media; geolocation; gyroscope; hid; microphone; midi; payment; usb; vr; xr-spatial-tracking"
+     sandbox="allow-forms allow-modals allow-popups allow-presentation allow-same-origin allow-scripts"
+   ></iframe>
+
+Note that the template stays the same, but we've extracted the logic to use in our script section differently. If, from here, we wanted to use `isShowing` in the Options API, we could access it with `this.isShowing` now, just as we normally do with a data property, and similarly, we can access `this.toggleShow` like we would a method.
+
+You may notice we used `reactive` here instead of `refs`. This is intentional- if you're refactoring a codebase, potentially full of many many values, `reactive` translates faster and more directly as the API is closer to Options, you're still using that same object notation.
+
+## In Place of Vuex
+
+It is possible to use the Composition API in place of Vuex, and save yourself a dependency. That said, it's not exactly necessary, and like many approaches, there are tradeoffs.
+
+If you're using Vuex, it's very clear exactly what centralized state is being used across the application. In addition, you gain the benefit of using a standardized pattern that comes with officially supported tooling in the DevTools which can make a big difference when it comes to debugging code.
+
+Composition API is very flexible, but you may lose that implicit declaration in communication to fellow maintainers. Our suggestion that if you do use it as a centralized state management store, that you place it in a `stateManagement` folder, or something similarly named, so that responsibilities are clear.
```

Comment:
I'd start this paragraph with *'The Composition API'*. It doesn't sound right to me without the *'The'*

Also, I think it needs *'is'* after the word *'suggestion'*.

Reply:
```suggestion
The Composition API is very flexible, but you may lose that implicit declaration in communication to fellow maintainers. Our suggestion is that if you do use it as a centralized state management store, that you place it in a `stateManagement` folder, or something similarly named, so that responsibilities are clear.
```